src/Behat/Testwork/ServiceContainer/Configuration/ConfigurationLoader.php

```diff
@@ -200,19 +200,25 @@ protected function loadFileConfiguration($configPath, $profile)
         $basePath = rtrim(dirname($configPath), DIRECTORY_SEPARATOR);
 
         if (str_ends_with($configPath, '.php')) {
-            $phpConfig = require $configPath;
-            if (!$phpConfig instanceof ConfigInterface) {
-                throw new ConfigurationLoadingException(sprintf('Configuration file `%s` must return an instance of `%s`.', $configPath, ConfigInterface::class));
-            }
-
-            $config = $phpConfig->toArray();
+            $config = $this->getPHPConfigObject($configPath)->toArray();
         } else {
             $config = (array) Yaml::parse(file_get_contents($configPath));
         }
 
         return $this->loadConfigs($basePath, $config, $profile);
     }
 
+    private function getPHPConfigObject(string $configPath): ConfigInterface
```

Comment:
@carlos-granados It prevents var collisions.
Now, the only possible collision is with "$configPath" in the throw exception.

Reply:
@loic425 I really don't understand why this change is needed, can you please explain? 🤔

Reply:
yeah indeed...

When using the "require" method. We override existing arguments from the current method (here the $profile var)

Reply:
Ahh, I get it now. Thanks!

Reply:
if you want to reduce conflict even more, you could make this private method a pure wrapper isolating `require` by returning `mixed`, and validating the ConfigInterface type in the caller.

Also, this should probably be a static method to avoid giving access to `$this` in the required file. Or maybe even stricter by using a bound closure to also remove access to `self` (and so also potential access to private state of the current class): https://github.com/composer/composer/blob/5eeba719d3f1388f68773c28e3971a7f441ee919/src/Composer/Autoload/ClassLoader.php#L575-L577

Reply:
@stof I've made the change, I hope this is exactly why you have described it.

Reply:
@carlos-granados Cause the PR is going to be harder than expected, maybe should I open another PR for this specific fix?